### DISCUSSION TITLE
Force Ideavim to use normal mode when switching from terminal back to editor

### DISCUSSION THREAD

**Discussion by wturrell** [JetBrains/ideavim#983] [General]
Hello. I'm using PhpStorm 2024.2.1 (PS-242.21829.154) and IdeaVim 2.16.0 (macOS Sonoma 14.6.1)

I have vim keybindings for bash enabled* in my `.bashrc` (`set -o vi`)

The problem is that whist, when opening the terminal, vim is always in insert mode (as I'd usually wish, because you usually just start typing), when you switch _back_ to the editor, it retains whichever mode you were in when you ran the command.  Often that's still insert.

This isn't great, it keeps throwing me because I expect to always be in normal when I start (or return) to working on a file, both in standalone vim and in the IDE.

Would be possible to fix / override / customise this? 

 * though see also: https://youtrack.jetbrains.com/issue/VIM-3543

**Comment by lippfi** [comment]
It should be fixed in the EAP (editor should return to NORMAL mode after switching back to editor) - https://youtrack.jetbrains.com/issue/VIM-3580/New-terminal-switches-editor-to-insert-mode
Could you please try the new EAP and tell if the issue is fixed?

**Reply by wturrell** [reply]
Thanks for the quick reply.  Spent about 15 minutes with the EAP - it's fixed, or at least _almost completely_ – a couple of times it didn't work, but to be honest as the new editor itself still feels buggy to me (it doesn't seem to recognise all keypresses and there are various problems with scrolling and focusing) I'm inclined to put it down to that...

**Reply by lippfi** [reply]
If you notice any specific scenario when it doesn't work, please let me know, I'll try to fix it

What do you mean by the "new editor"?

**Reply by wturrell** [reply]
Sorry, terminal, not editor – the one that's in beta. 
If I spot a pattern related to Ideavim I'll let you know.
